crates/wallet/src/wallet/tx_builder.rs

```diff
@@ -145,6 +143,34 @@ pub(crate) struct TxParams {
     pub(crate) allow_dust: bool,
 }
 
+impl Default for TxParams {
+    fn default() -> Self {
+        Self {
+            recipients: Vec::new(),
+            drain_wallet: false,
+            drain_to: None,
+            fee_policy: None,
+            internal_policy_path: None,
+            external_policy_path: None,
+            utxos: Vec::new(),
+            unspendable: HashSet::new(),
+            manually_selected_only: false,
+            sighash: None,
+            ordering: TxOrdering::default(),
+            locktime: None,
+            rbf: Some(RbfValue::Default),
```

Comment:
I think this would be cleaner if we had `RbfValue` implement Default. Or we can let it default to None as it does now and handle the option when creating the tx, but that's not my first choice.

Reply:
If `RbfValue` does impl `Default`, this field will still [default to `None`](https://doc.rust-lang.org/std/option/enum.Option.html#impl-Default-for-Option%3CT%3E).

I'm thinking to do the following:

```rust
// Default is none.
sequence_policy: Option<SequencePolicy>

enum SequencePolicy {
    SignalNoRbf,
    Exact(Sequence),
}
```

`SequencePolicy::SignalNoRbf` Behavior:

* Set `nSequence` to `0xFFFFFFFE` if we have `nLockTime` set.
* Set `nSequence` to `0xFFFFFFFF` if we do not have `nLockTime` set.
* If this conflicts with `TxParams::locktime` or spk satisfaction requirements, error out.

`SequencePolicy::Exact(_)`:

* `nSequence` will be this value.
* If spk has `nSequence` requirements, and this value does not satisfy it, error out.

Reply:
I think it makes sense to have corresponding methods `.signal_no_rbf` and `.set_exact_sequence`.

Reply:
If no `SequencePolicy` is defined, and spk has no `OP_CHECKSEQUENCEVERIFY`, default to `0xFFFFFFFD` imo.

Reply:
Yes, we want to be able to derive Default for TxParams, so why not make `RbfValue` (or whatever we decide to call it) into a tuple struct that holds a `Sequence` and has a default value of 0xFFFFFFFD. Then the field of TxParams can just be `rbf: RbfValue` instead of an Option.

Reply:
I agree, my implementation of Default is too verbose for what it accomplishes. 

evanjlin: you mean removing `enable_rbf`, `enable_rbf_with_sequence` and `disable_rbf`, and then adding `signal_no_rbf` and `set_exact_sequence`?

ValuedMammal: I think this is a good approach. I'm inclined to go with it, unless we plan to change anything else that touches this part of the code.

Reply:
> Yes, we want to be able to derive Default for TxParams, so why not make `RbfValue` (or whatever we decide to call it) into a tuple struct that holds a `Sequence` and has a default value of 0xFFFFFFFD. Then the field of TxParams can just be `rbf: RbfValue` instead of an Option.

So by default we want to set `nSequence` to `0xFFFFFFFD` (according to the `RbfValue` default). What if the SPK we are spending contains a relative timelock? Who do we listen to? Our default `RbfValue` setting? Or the SPK?

Reply:
@ValuedMammal I think it makes sense to have `RbfValue` (or whatever we name it) be wrapped in `Option`, where `None` is the default (by default).

`None` would just mean set to the default enable RBF flag (`0xFFFFFFFD`), unless if a RTL is needed as according to the SPK we are spending.

Reply:
I struggle to think of a reason why a user would actually need a method like `disable_rbf` when we have `set_exact_sequence` (or what I'm calling `nsequence` in keeping with tradition). With this PR we will automatically be compliant with nodes that enforce explicit signaling under bip125. To the rest of the network, this signaling mechanism via the nSequence field is not relevant for evaluating replacements (AFAIK). Hence my claim that neither `enable_rbf` nor `disable_rbf` are necessary parts of the API. I agree that if the descriptor requires an absolute locktime, then we should disallow setting `Sequence::MAX`.